sklearn/neighbors/_graph.py

```diff
@@ -35,6 +44,19 @@ def _query_include_self(X, include_self, mode):
     return X
 
 
+@validate_params(
+    {
+        "X": ["array-like", BallTree, KNeighborsMixin],
+        "n_neighbors": [Interval(Integral, 1, None, closed="left")],
+        "mode": [StrOptions({"connectivity", "distance"})],
+        "metric": [StrOptions(set(itertools.chain(*VALID_METRICS.values()))), callable],
+        "p": [Interval(Real, 0, None, closed="right"), None],
+        "metric_params": [dict, None],
+        "include_self": ["boolean", StrOptions({"auto"})],
+        "n_jobs": [Integral, None],
+    },
+    prefer_skip_nested_validation=True,
```

Comment:
Sorry, I don’t understand. 

The inner validation comes into play if `X` is made a `NearestNeighbors` estimator by the function, correct? 
And we shouldn’t skip, because `metric` hasn’t been validated yet?